### ISSUE TITLE
README instructions don't work

### ISSUE THREAD

**Issue by ebkalderon** [Profpatsch/yarn2nix#25]
There are a few issues with the `default.nix` described in the README currently:

```nix
let
  nixpkgsPath = <nixpkgs>;
  pkgs = import nixpkgsPath {};
  nixLib = pkgs.callPackage /path/to/yarn2nix/nix-lib {
    # WARNING (TODO): for now you need to use this checked out yarn2nix
    # because the upstream package (in haskellPackages) might have
    # broken dependencies (yarn-lock and yarn2nix are not in stackage)
    yarn2nix = import /path/to/yarn2nix { inherit nixpkgsPath; };
  };

in
  nixLib.buildNodePackage
    ( { src = nixLib.removePrefixes [ "node_modules" ] ./.; } //
      nixLib.callTemplate ./npm-package.nix
        (nixLib.buildNodeDeps (pkgs.callPackage ./npm-deps.nix {}))
```

1. There is an unmatched closing parentheses missing at the end, which is an easy enough fix.
2. I think that d607336935c8131872ee97a7f12aa7a0965f63f6 wasn't tested properly with the README, because it is no longer possible to call `buildNodePackage` with the key/value pairs produced by the generated `npm-package.nix`, as shown in the instructions, because `name` was replaced with `key { scope = ""; name = ""; }`.

**Comment by sternenseemann** [comment]
Indeed <https://github.com/Profpatsch/yarn2nix/commit/d607336935c8131872ee97a7f12aa7a0965f63f6> introduced a bug because `yarn2nix --template` wasn't updated to produce a `key` attribute.

Possible fixes:

* Make `yarn2nix --template` produce a `key` attribute
* Accept a `name` attr in `buildNodePackage` 

The intended use (as in README example), currently results in the following error, since `--template` produces `name` instead of `key`.

```
error: anonymous function at /home/lukas/src/nix/yarn2nix/nix-lib/buildNodePackage.nix:2:1 called without required argument 'key', at /home/lukas/src/nix/yarn2nix/test.nix:9:3
```

**Comment by sternenseemann** [comment]
Fixed in 1ad49b3bd144f2ddba8d884fe34d63f108120d21 and f04bdf6641b27e9f2af935c22701739609f0c558.
